File: testers/avro/java-quarkus/generated-and-checked-in/com/example/events/OrderEventsPublisher.java
package com.example.events;

import io.smallrye.mutiny.Uni;
import io.smallrye.reactive.messaging.MutinyEmitter;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;
import org.eclipse.microprofile.reactive.messaging.Channel;

/** Type-safe event publisher for order-events topic */
@ApplicationScoped
public record OrderEventsPublisher(
    @Channel("order-events") MutinyEmitter<OrderEvents> kafkaTemplate, String topic) {
  @Inject
  public OrderEventsPublisher {}

  public OrderEventsPublisher(@Channel("order-events") MutinyEmitter<OrderEvents> kafkaTemplate) {
    this(kafkaTemplate, "order-events");
  }

  public OrderEventsPublisher withKafkaTemplate(MutinyEmitter<OrderEvents> kafkaTemplate) {
    return new OrderEventsPublisher(kafkaTemplate, topic);
  }

  public OrderEventsPublisher withTopic(String topic) {
    return new OrderEventsPublisher(kafkaTemplate, topic);
  }

  /** Publish a OrderCancelled event */
  public Uni<Void> publish(String key, OrderCancelled event) {
    return kafkaTemplate.send(event);
  }

  /** Publish a OrderPlaced event */
  public Uni<Void> publish(String key, OrderPlaced event) {
    return kafkaTemplate.send(event);
  }

  /** Publish a OrderUpdated event */
  public Uni<Void> publish(String key, OrderUpdated event) {
    return kafkaTemplate.send(event);
  }
}

File: testers/avro/kotlin-quarkus-mutiny/generated-and-checked-in/com/example/events/OrderEventsPublisher.kt
package com.example.events

import io.smallrye.mutiny.Uni
import io.smallrye.reactive.messaging.MutinyEmitter
import jakarta.enterprise.context.ApplicationScoped
import jakarta.inject.Inject
import java.lang.Void
import org.eclipse.microprofile.reactive.messaging.Channel

@ApplicationScoped
/** Type-safe event publisher for order-events topic */
data class OrderEventsPublisher @Inject constructor(
  @field:Channel("order-events") val kafkaTemplate: MutinyEmitter<OrderEvents>,
  val topic: kotlin.String = "order-events"
) {
  /** Publish a OrderCancelled event */
  fun publish(
    key: kotlin.String,
    event: OrderCancelled
  ): Uni<Void> {
    return kafkaTemplate.send(event)
  }

  /** Publish a OrderPlaced event */
  fun publish(
    key: kotlin.String,
    event: OrderPlaced
  ): Uni<Void> {
    return kafkaTemplate.send(event)
  }

  /** Publish a OrderUpdated event */
  fun publish(
    key: kotlin.String,
    event: OrderUpdated
  ): Uni<Void> {
    return kafkaTemplate.send(event)
  }
}